### DISCUSSION TITLE
Unable to save cookies in iOS

### DISCUSSION THREAD

**Discussion by Carben-dev** [ionic-team/capacitor#3537] [Q&A]
I am using capacitor (https://capacitorjs.com/) to wrap my angular frontend. The first login screen can load without a problem.
But I find out the angular frontend is not able to save any cookies. 

The capacitor wrapper is working fine on Android. Cookies can be saved normally. This only happened on iOS. (iOS version use capacitor:// as it's protocol, which makes the OS thinks it's a different domain??)

I find a discussion about share cookies on cross-domain: https://github.com/ionic-team/capacitor/issues/1373, but it looks like no solution on that topic yet.

Does anyone have a solution to that?

Thank you.

**Comment by filipegeric** [comment]
Adding this to `capacitor.config.json` helped me with this issue:
```json
{
  ...
  "server": {
    "hostname": "API_DOMAIN"
  }
}
```

*Replace API_DOMAIN with your domain (api.example.com)

**Comment by tafelnl** [comment]
The solution of @filipegeric indeed helps with server-side cookies. These will be stored and sent out when necessary. But you still cannot set or access cookies from the client-side. Besides that, this solution should only be deployed for iOS and not for Android. On Android you should keep `http://localhost` as cookies work just fine on Android already and changing `http://localhost` to something else will break the secure context.

**Comment by tafelnl** [comment]
Please check my elaborate answer here: https://github.com/ionic-team/capacitor/issues/1373#issuecomment-707822708
